tests/unit/lib/redis-nodetest.js

```diff
@@ -80,25 +80,71 @@ describe('redis', function() {
         });
     });
 
-    it('trims the list of recent uploads', function() {
-      var recentUploads = ['a', 'b', 'c'];
+    it('trims the list of recent uploads and removes the index key', function() {
+      var recentUploads = ['1','2','3','4','5','6','7','8','9','10','11'];
+      var finalUploads  = ['3','4','5','6','7','8','9','10','11','12'];
 
       var redis = new Redis({}, new FakeRedis(FakeClient.extend({
         get: function(key) {
           return Promise.resolve(null);
         },
-        lpush: function(key, tag) {
-          recentUploads.push(key + tag);
+        set: function(key, value) {
         },
-        ltrim: function() {
-          recentUploads.pop();
+        zadd: function(key, score, revisionKey) {
+          recentUploads.push(revisionKey);
+        },
+        zrem: function(val,revision) {
+          var i = recentUploads.indexOf(revision)
```

Comment:
yup, that's probably not a bad idea, right now I rely on `recentUploads` being closed upon to run my tests though, that said it can probably be tweaked/improved to reduce repetition